apps/verticalhome/js/app.js

```diff
@@ -157,14 +204,43 @@
       this.renderGrid();
     },
 
+
+    /**
+     * We need to do some refreshing thing after l10n is ready.
+     *
+     * @memberof LockScreen
+     * @this {LockScreen}
+     */
+    l10nInit: function ls_l10nInit() {
+      this.l10nready = true;
+      // The default one is 12 hour.
+      this.timeFormat = window.navigator.mozHour12 ?
+        navigator.mozL10n.get('shortTimeFormat12') :
+        navigator.mozL10n.get('shortTimeFormat24');
+      this.refreshClock(new Date());
+
+    },
+
+
+    refreshClock: function ls_refreshClock(now) {
+      var f = new navigator.mozL10n.DateTimeFormat();
+      var _ = navigator.mozL10n.get;
+
+      var timeFormat = this.timeFormat.replace('%p', '<span>%p</span>');
+      var dateFormat = _('longDateFormat');
```

Comment:
It doesn't seem right to do localization over and over again every minute. That's inefficient. Please separate the code for updating the formats from the code for updating the locales.

And if you can, move this code into your clock.js module.